lib/azure/core/utility.rb

```diff
@@ -0,0 +1,244 @@
+# Copyright 2013 Microsoft Open Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#--------------------------------------------------------------------------
+
+require 'ipaddr'
+
+if RUBY_VERSION.to_f < 2.0
+  begin
+    require 'Win32/Console/ANSI' if RUBY_PLATFORM =~ /win32|mingw32/
+  rescue LoadError
+    puts 'WARNING: Output will look weird on Windows unless'\
+         ' you install the "win32console" gem.'
+  end
+end
+
+module Azure
+  module Error
+    # Azure Error
+    class Error <  Azure::Core::Error
+      attr_reader :description
+      attr_reader :status_code
+      attr_reader :type
+
+      def initialize(type, status, description)
+        @type = type
+        @status_code = status
+        @description = description
+        super("#{type} (#{status_code}): #{description}")
+      end
+    end
+  end
+
+  module Core
+    module Utility
+      def random_string(str = 'azure', no_of_char = 5)
+        str + (0...no_of_char).map { ('a'..'z').to_a[rand(26)] }.join
+      end
+
+      def xml_content(xml, key, default = '')
+        content = default
+        node = xml.at_css(key)
+        content = node.text if node
+        content
+      end
+
+      def locate_file(name)
+        if File.exist? name
+          name
+        elsif File.exist?(File.join(ENV['HOME'], name))
+          File.join(ENV['HOME'], name)
+        else
+          Azure::Loggerx.error_with_exit "Unable to find #{name} file  "
+        end
+      end
+
+      def export_der(cert, key, pass = nil, name = nil)
+        pkcs12 = OpenSSL::PKCS12.create(pass, name, key, cert)
+        Base64.encode64(pkcs12.to_der)
+      rescue Exception => e
+        puts e.message
+        abort
+      end
+
+      def export_fingerprint(certificate)
+        Digest::SHA1.hexdigest(certificate.to_der)
+      end
+
+      def enable_winrm?(winrm_transport)
+        (!winrm_transport.nil? && (winrm_transport.select { |x| x.downcase == 'http' || x.downcase == 'https' }.size > 0))
+      end
+
+      def get_certificate(private_key_file)
+        rsa = OpenSSL::PKey.read File.read(private_key_file)
+        cert = OpenSSL::X509::Certificate.new
+        cert.version = 2
+        cert.serial = 0
+        name = OpenSSL::X509::Name.new([['CN', 'Azure Management Certificate']])
+        cert.subject = cert.issuer = name
+        cert.not_before = Time.now
+        cert.not_after = cert.not_before + (60*60*24*365)
```

Comment:
Minor: this code may not pass the leap year code scan. If we don't care the cert validity is exactly one year, we could consider other value.